packages/next/src/build/webpack-config.ts

```diff
@@ -1309,6 +1309,12 @@ export default async function getBaseWebpackConfig(
                     },
                   ],
                 },
+                resourceQuery: {
+                  not: [
+                    new RegExp(WEBPACK_RESOURCE_QUERIES.metadata),
```

Comment:
These 2 sounds fine to keep as RSC like layers, curious why we need to change them here?

Reply:
They will still be in the RSC layer (via issuer layer), but we don't want to apply the `next-flight-loader` to them, as that would generate additional and conflicting entries, which are already covered by the `next-metadata-route-loader` entries.

Reply:
Without this exclusion, we would generate duplicate export statements like these:
```
export { $$RSC_SERVER_CACHE_0 as "8056ace5d0b6cd298fba2780df50a7ec233c33eeef" } from "<snip>/app/opengraph-image.tsx?__next_metadata__"
export { $$RSC_SERVER_CACHE_0 as "8056ace5d0b6cd298fba2780df50a7ec233c33eeef" } from "<snip>/app/opengraph-image.tsx?__next_metadata_image_meta__"
```

I've pushed a (hopefully) clarifying comment.